ai-decision-services/ai-ad-dpp-engine/prepare-config-data-and-app/prepare-config-data-and-app.md

```diff
@@ -0,0 +1,217 @@
+*   Library (Archive.zip)
+
+## Orchestration Configuration
+
+This is a JSON file that defines the DPP workflow. The configuration is composed of the following 5 parts:
+
+<table class="wrapped confluenceTable"><colgroup><col><col></colgroup><tbody><tr><th class="confluenceTh">Section Name</th><th class="confluenceTh">Explanation</th></tr><tr><td class="confluenceTd">inputSources</td><td class="confluenceTd">Connector for input data sources. Currently we support reading data from Object Storage, ATP and ADW.</td></tr><tr><td class="highlight-blue confluenceTd" data-highlight-colour="blue">phaseInfo<td class="highlight-blue confluenceTd" data-highlight-colour="blue">A flag specifies whether it's training or inferencing, and connector to metadata source.</td></tr><tr><td class="confluenceTd">processingSteps</td><td class="confluenceTd">Data preprocessing transformers and related input arguments.</td></tr><tr><td class="highlight-blue confluenceTd" data-highlight-colour="blue">stagingDestination</td><td class="highlight-blue confluenceTd" data-highlight-colour="blue">Connector for intermediate processed data storage, which will be used in post-processing steps.</td></tr><tr><td class="confluenceTd">outputDestination</td><td class="confluenceTd">Connector for final output.</td></tr><tr><td class="highlight-blue confluenceTd" data-highlight-colour="blue">serviceApiConfiguration (Optional)</td><td class="highlight-blue confluenceTd" data-highlight-colour="blue">Configuration required for post-processing steps. For example, arguments for anomaly detection client.</td></tr></tbody></table>
+
+## inputSources
+
+Dataset(s) input source(s), with **inputSources** as key and an array of dataset(s) information as value.
+
+*   **dataframeName** - the variable name of this data frame will be used in the whole context.
+*   **type** - can either be **object-storage** or **oracle**.
+*   **isEventDriven** - (optional) This is an optional parameter whose default value is _false_. It takes a boolean value of _true_ is provided if the input source needs to be taken from a configured event. Note that currently, it is only supported for Object Storage emitted events.
+*   For object storage, you will require:
+    *  **namespace** \- the namespace of the target object storage bucket. 
+    *   **bucket** - the bucket name. 
+    *   **objectName** \- the full path and name of the object. This can also be a prefix with star (\*) when _isEventDriven_ is provided with _true_ value, where the object generating the event is considered and attempted to read if the name matches the provided expression. For example, updating object with name _anomalies/test.csv_ can generate an event and can be captured by using _anomalies/\*_ as the objectName, provided isEventDriven is set to true. Also, note that anything after \* is ignored in our current version and we will add more functionality in upcoming sprints.
+*   For oracle, you will require:
```

Comment:
I think we should remove instructions for ADW from here as it is not applicable to this workshop. It can be left in the configuration doc,

Reply:
@ganrad suggested me to move the configuration doc completely to here. Here is the finalized place for introducing how to write a configuration, so I need to have ADW info too

Reply:
Sure, we can leave it here. BTW I would recommend against adding ADW data source for the more complex example since it can be hard to explain the data upload steps